bgw-core/src/main/kotlin/tools/aqua/bgw/components/uicomponents/TextArea.kt

```diff
@@ -42,17 +42,25 @@ import tools.aqua.bgw.util.Font
  *  @see TextField
  */
 open class TextArea(
-	posX: Number = 0,
-	posY: Number = 0,
-	width: Number = DEFAULT_TEXT_AREA_WIDTH,
-	height: Number = DEFAULT_TEXT_AREA_HEIGHT,
-	text: String = "",
-	font: Font = Font(),
-	val prompt: String = "",
+    posX: Number = 0,
+    posY: Number = 0,
+    width: Number = DEFAULT_TEXT_AREA_WIDTH,
+    height: Number = DEFAULT_TEXT_AREA_HEIGHT,
+    text: String = "",
+    font: Font = Font(),
+    prompt: String = "",
+    internal var onUpdatePrompt: ((prompt: String) -> Unit)? = null,
```

Comment:
This should be implemented via an observable property and not with a function reference.